builder/vsphere/driver/vm.go

```diff
@@ -351,52 +351,56 @@ func (vm *VirtualMachineDriver) Clone(ctx context.Context, config *CloneConfig)
 
 	virtualDisks := devices.SelectByType((*types.VirtualDisk)(nil))
 	virtualControllers := devices.SelectByType((*types.VirtualController)(nil))
+	virtualNetworkAdapters := devices.SelectByType((*types.VirtualEthernetCard)(nil))
 
 	// Use existing devices to avoid overlapping configuration
 	existingDevices := object.VirtualDeviceList{}
 	existingDevices = append(existingDevices, virtualDisks...)
 	existingDevices = append(existingDevices, virtualControllers...)
+	existingDevices = append(existingDevices, virtualNetworkAdapters...)
 
 	storageConfigSpec, err := config.StorageConfig.AddStorageDevices(existingDevices)
 	if err != nil {
 		return nil, fmt.Errorf("failed to add storage devices: %s", err.Error())
 	}
 	configSpec.DeviceChange = append(configSpec.DeviceChange, storageConfigSpec...)
 
-	if config.Network != "" {
-		net, err := vm.driver.FindNetwork(config.Network)
-		if err != nil {
-			return nil, fmt.Errorf("Error finding network: %s", err)
-		}
-		backing, err := net.network.EthernetCardBackingInfo(ctx)
-		if err != nil {
-			return nil, fmt.Errorf("Error finding ethernet card backing info: %s", err)
-		}
-
-		devices, err := vm.vm.Device(ctx)
-		if err != nil {
-			return nil, fmt.Errorf("Error finding vm devices: %s", err)
+	networkNames := splitTrim(config.Network)
+	if len(networkNames) > 0 {
+		if len(networkNames) > len(virtualNetworkAdapters) {
+			return nil, fmt.Errorf("%d network names defined but only %d network adapters exist", len(networkNames), len(virtualNetworkAdapters))
 		}
 
-		adapter, err := findNetworkAdapter(devices)
-		if err != nil {
-			return nil, fmt.Errorf("Error finding network adapter: %s", err)
-		}
+		macAddresses := splitTrim(config.MacAddress)
+		for i, networkName := range networkNames {
+			// empty network name means don't touch it
+			if networkName == "" {
+				continue
+			}
 
-		current := adapter.GetVirtualEthernetCard()
-		current.Backing = backing
+			net, err := vm.driver.FindNetwork(networkName)
+			if err != nil {
+				return nil, fmt.Errorf("Error finding network: %s", err)
+			}
+			backing, err := net.network.EthernetCardBackingInfo(ctx)
+			if err != nil {
+				return nil, fmt.Errorf("Error finding ethernet card backing info: %s", err)
+			}
 
-		if config.MacAddress != "" {
-			current.AddressType = string(types.VirtualEthernetCardMacTypeManual)
-			current.MacAddress = config.MacAddress
-		}
+			adapter := virtualNetworkAdapters[i].(types.BaseVirtualEthernetCard)
+			card := adapter.GetVirtualEthernetCard()
+			card.Backing = backing
+			if len(macAddresses) > i && macAddresses[i] != "" {
```

Comment:
This type of validation to ensure that the length of macAddresses is the same as the length of networkNames should be handled during the prepare step. To avoid any potential misconfigurations